src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java

```diff
@@ -125,4 +132,76 @@ public void verifySaveIntegrationPluginOfType_postsExpectedRequestBody() throws
+     * The values defined as part of the request matcher are what is expected to be set due to the users configuration.
+     * @throws JsonProcessingException
+     */
+    @Test
+    public void verifySaveIntegrationPluginOfType_postsExpectedRequestBodyWithBasePath() throws JsonProcessingException {
+        swaggerHubClient = buildSwaggerHubClient("basePath");
+        //Given
+        String owner = "owner";
+        String api = "api";
+        String version = "1.0.0";
+        String oas = "3.0.0";
+        String branch = "branch-test";
+        boolean enabled = true;
+        String outputFile = "outputFile";
+        String repository = "repository";
+        String repositoryOwner = "repositoryOwner";
+        String scmProvider = "GITHUB";
+        String token = "token";
+        String target = "JSON (Unresolved)";
+        String syncMethod = "Advanced Sync";
+        String name = "Integration Name";
+        String outputFolder = "output/folder";
+
+        SaveSCMPluginConfigRequest.Builder requestBuilder = new SaveSCMPluginConfigRequest.Builder(owner, api, version)
+                .oas(oas)
+                .branch(branch)
+                .enabled(enabled)
+                .outputFile(outputFile)
+                .repository(repository)
+                .repositoryOwner(repositoryOwner)
+                .scmProvider(scmProvider)
+                .target(target)
+                .token(token)
+                .outputFolder(outputFolder)
+                .managedPaths(new String[]{outputFile})
+                .name(name);
+
+        SaveSCMPluginConfigRequest saveSCMPluginConfigRequest = requestBuilder.build();
+
+        String requestUrl = String.format("/basePath/plugins/configurations/%s/%s/%s/%s?oas=%s", owner, api, version, scmProvider, oas);
+        stubFor(put(requestUrl).willReturn(created()));
+
+        RequestPatternBuilder putRequestPattern = putRequestedFor(urlPathEqualTo(String.format("/basePath/plugins/configurations/%s/%s/%s/%s", owner, api, version, scmProvider)))
+                .withQueryParam("oas", equalTo(oas))
+                .withRequestBody(matchingJsonPath("$.branch", equalTo(branch)))
+                .withRequestBody(matchingJsonPath("$.enabled", equalTo("true")))
+                .withRequestBody(matchingJsonPath("$.outputFile", equalTo(outputFile)))
+                .withRequestBody(matchingJsonPath("$.repository", equalTo(repository)))
+                .withRequestBody(matchingJsonPath("$.owner", equalTo(repositoryOwner)))
+                .withRequestBody(matchingJsonPath("$.syncMethod", equalTo(syncMethod)))
+                .withRequestBody(matchingJsonPath("$.target", equalTo(target)))
+                .withRequestBody(matchingJsonPath("$.outputFolder", equalTo(outputFolder)))
+                .withRequestBody(matchingJsonPath("$.managedPaths", equalToJson("[\""+outputFile+"\"]")))
+                .withRequestBody(matchingJsonPath("$.name", equalTo(name)));
```

Comment:
Might be no harm to absract this out to a common method that can be used by each test. For the most part the matcher pattern is the same.

Reply:
I have put this into an abstract method now. We can discuss maybe a better way of doing this as the tests now will be very hard to update if we add more tests to cover more functionality